private/private.go

```diff
@@ -46,55 +49,183 @@ type PrivateTransactionManager interface {
+	var err error
+	P, err = NewPrivateTxManager(cfg)
+	return err
+}
+
+func IsQuorumPrivacyEnabled() bool {
+	return isPrivacyEnabled
+}
+
+func NewPrivateTxManager(cfg engine.Config) (PrivateTransactionManager, error) {
+
+	if cfg.ConnectionType == engine.NoConnection {
+		log.Info("Running with private transaction manager disabled - quorum private transactions will not be supported")
+		return &notinuse.PrivateTransactionManager{}, nil
 	}
 
-	client := &engine.Client{
-		HttpClient: &http.Client{
-			Transport: unixTransport(cfg),
-		},
-		BaseURL: "http+unix://c",
+	client, err := createClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create connection to private tx manager due to: %s", err)
 	}
 
 	ptm, err := selectPrivateTxManager(client)
 	if err != nil {
-		return nil, fmt.Errorf("unable to connect to private tx manager using %s due to %s", cfgPath, err)
+		return nil, fmt.Errorf("unable to connect to private tx manager due to: %s", err)
 	}
+
+	isPrivacyEnabled = true
 	return ptm, nil
 }
 
+func createClient(cfg engine.Config) (*engine.Client, error) {
+	var client *engine.Client
+	if engine.IsSocketConfigured(cfg) {
+
+		log.Info("Connecting to private tx manager using IPC socket")
+		client = &engine.Client{
+			HttpClient: &http.Client{
+				Transport: unixTransport(cfg),
+			},
+			BaseURL: "http+unix://c",
+		}
+
+	} else {
+
+		transport := httpTransport(cfg)
+		if cfg.TlsMode == engine.TlsOff {
+			log.Info("Connecting to private tx manager using HTTP")
+		} else {
+			log.Info("Connecting to private tx manager using HTTPS")
+			err := setHttpTransportToUseTLS(cfg, transport)
+			if err != nil {
+				return nil, fmt.Errorf("unable to create http.client to private tx manager due to: %s", err)
+			}
+		}
+
+		client = &engine.Client{
+			HttpClient: &http.Client{
+				Timeout:   time.Duration(cfg.Timeout) * time.Second,
+				Transport: transport,
+			},
+			BaseURL: cfg.HttpUrl,
+		}
+
+	}
+
+	return client, nil
+}
+
 func unixTransport(cfg engine.Config) *httpunix.Transport {
+	// Note that clientTimeout doesn't work when using httpunix.Transport, so we set ResponseHeaderTimeout instead
 	t := &httpunix.Transport{
 		DialTimeout:           time.Duration(cfg.DialTimeout) * time.Second,
-		RequestTimeout:        time.Duration(cfg.RequestTimeout) * time.Second,
-		ResponseHeaderTimeout: time.Duration(cfg.ResponseHeaderTimeout) * time.Second,
+		RequestTimeout:        5 * time.Second,
+		ResponseHeaderTimeout: time.Duration(cfg.Timeout) * time.Second,
 	}
 	t.RegisterLocation("c", filepath.Join(cfg.WorkDir, cfg.Socket))
 	return t
 }
 
+func httpTransport(cfg engine.Config) *http.Transport {
+	t := &http.Transport{
+		IdleConnTimeout: time.Duration(cfg.HttpIdleConnTimeout) * time.Second,
+		WriteBufferSize: cfg.HttpWriteBufferSize,
+		ReadBufferSize:  cfg.HttpReadBufferSize,
+	}
+	return t
+}
+
+func setHttpTransportToUseTLS(cfg engine.Config, transport *http.Transport) error {
```

Comment:
I think this function would be more convenient with next signature `NewTLSConfig(cfg engine.Config) (*tls.Config, error)` (only generating the tls.Config from configuration). 

It can then be easily re-used by upper-level functions creating the http.Transport

Reply:
Done